core/output/facebook.js

```diff
@@ -4,17 +4,61 @@ var fb = require("facebook-chat-api"),
 	stopListeningMethod = null,
 	platform = null,
 	endTyping = null,
-	platformApi = null;
+	platformApi = null,
+	unknownIter = 1,
+	threadInfo = {};
+
+var getSenderInfo = function (ids, api, event, finished) {
+	var callback = function(err, info) {
+		if (err) {
+			return finished('<Unknown User ' + unknownIter++ + '>');
+		}
+		for (var id in info) {
+			threadInfo[event.threadID][id] = {
+				name: info[id].name,
+				email: 'unknown@foo.bar'
+			};
+		}
+		return finished(threadInfo[event.threadID][event.senderID].name);
+	};
+	api.getUserInfo(ids, callback);
+}
+
+var getSenderName = function(api, event, finished) {
+	if (threadInfo[event.threadID] && threadInfo[event.threadID][event.senderID]) {
+		return finished(threadInfo[event.threadID][event.senderID].name);
+	}
+
+	if (!threadInfo[event.threadID]) {
+		threadInfo[event.threadID] = {};
+		api.getThreadInfo(event.threadID, function(err, info) {
+			if (err) {
+				return finished('<Unknown User ' + unknownIter++ + '>');
+			}
+			getSenderInfo(info.participantIDs, api, event, finished);
+		});
+	}
+	else {
+		getSenderInfo([event.senderID], api, event, finished);
+	}
+};
 
 exports.start = function(callback) {
 	fb({email: this.config.username, password: this.config.password}, function (err, api) {
 		if(err) {
 			console.error(err);
 			process.exit(-1);
 		}
-		api.setOptions({listenEvents: true});
+
+		var options = {
+			listenEvents: true
```

Comment:
Any reason why `options.logLevel` is not added inline using lambdas as 
`logLevel: console.isDebug() ? 'verbose' : 'silent' `